src/libponyrt/gc/serialise.c

```diff
@@ -186,15 +196,22 @@ PONY_API size_t pony_serialise_offset(pony_ctx_t* ctx, void* p)
   if(s != NULL)
   {
     if(s->block || (s->t != NULL && s->t->serialise != NULL))
-      return s->value;
+      return (uintptr_t)s->value;
     else
       return ALL_BITS;
   }
 
-  // If we are not in the map, we are an untraced primitive. Return the type id
-  // with the high bit set.
+  // TODO: FIND OUT WHEN/HOW THIS CAN HAPPEN!!!!!!
+  // If not possible normally, make this into an error/assert!
+  // If possible, figure out how to handle 64 bit serialise_id's and return types
+  // since the pointers are no longer able to hold the serialise_id + high bit
+  // twiddles.
+  // See TODO in pony_deserialise_offset for related issue
+
+  // If we are not in the map, we are an untraced primitive. Return the
```

Comment:
@sylvanc would you be able to provide input on when this code path might be taken? (it would also be great if you were able to review this PR overall)